netty/src/main/java/io/grpc/netty/NettyClientStream.java

```diff
@@ -105,113 +115,133 @@ public void start(ClientStreamListener listener) {
       public void operationComplete(ChannelFuture future) throws Exception {
         if (!future.isSuccess()) {
           // Stream creation failed. Close the stream if not already closed.
-          Status s = statusFromFailedFuture(future);
-          transportReportStatus(s, true, new Metadata());
+          Status s = transportState().statusFromFailedFuture(future);
+          transportState().transportReportStatus(s, true, new Metadata());
         }
       }
     };
 
     // Write the command requesting the creation of the stream.
-    writeQueue.enqueue(new CreateStreamCommand(http2Headers, this),
+    writeQueue.enqueue(new CreateStreamCommand(http2Headers, transportState()),
         !method.getType().clientSendsOneMessage()).addListener(failureListener);
   }
 
-  @Override
-  public void transportReportStatus(Status newStatus, boolean stopDelivery, Metadata trailers) {
-    super.transportReportStatus(newStatus, stopDelivery, trailers);
-  }
+  private class Sink implements AbstractClientStream2.Sink {
+    @Override
+    public void writeFrame(WritableBuffer frame, boolean endOfStream, boolean flush) {
+      ByteBuf bytebuf = frame == null ? EMPTY_BUFFER : ((NettyWritableBuffer) frame).bytebuf();
```

Comment:
Meh. There is only one writer to this, `MessageFramer`, and it doesn't hurt anything, as you said. The restriction is more to benefit this class (by reducing the cases it must handle) than to require it to have additional checks. I'd favor changing the method specification than adding the extra check.